TypeScript error due to missing `Performance` type in Node environments (since v15.3)

Starting with version `15.3.2`, `@sinonjs/fake-timers` introduces a TypeScript compilation error in Node.js-only environments:

```ts
Error: node_modules/@sinonjs/fake-timers/types/fake-timers-src.d.ts(109,19): error TS2304: Cannot find name 'Performance'.
```

###  Environment

* `@sinonjs/fake-timers`: ≥ 15.3.2
* TypeScript: reproducible with recent versions
* Runtime: Node.js (no DOM)

### Steps to reproduce

1. Use `@sinonjs/fake-timers` in a TypeScript project
2. Configure `tsconfig.json` without DOM types:

```json
{
  "compilerOptions": {
    "lib": ["ES2022"]
  }
}
```

3. Run the TypeScript compiler

###  Actual behavior

The compiler fails because `Performance` is referenced in the type definitions but not available in a Node.js-only type environment.

###  Expected behavior

The package should compile in a Node.js environment without requiring DOM types.

###  Workarounds

The issue can currently be worked around by:

* Enabling `skipLibCheck`:

```json
{
  "compilerOptions": {
    "skipLibCheck": true
  }
}
```

* Adding `"DOM"` to `lib`:

```json
{
  "compilerOptions": {
    "lib": ["ES2022", "DOM"]
  }
}
```

###  Why this is a problem

* `skipLibCheck` disables type checking for all dependencies and may hide real issues.
* Adding `"DOM"` introduces browser-specific globals into a Node.js project, which is undesirable and semantically incorrect.

--- a/package.json
+++ b/package.json
@@ -33,7 +33,7 @@
     "prepare": "husky",
     "types:build": "tsgo -p tsconfig.types.json",
     "types:check": "tsgo -p tsconfig.types.json --noEmit",
-    "types:smoke": "tsgo -p test/typescript-consumer/tsconfig.json --noEmit"
+    "types:smoke": "tsgo -p test/typescript-consumer/tsconfig.json --noEmit && tsgo -p test/typescript-no-dom-consumer/tsconfig.json --noEmit"
   },
   "types": "./types/fake-timers-src.d.ts",
   "lint-staged": {

--- a/src/fake-timers-src.js
+++ b/src/fake-timers-src.js
@@ -307,6 +307,10 @@ if (typeof require === "function" && typeof module === "object") {
  */
 
 /**
+ * @typedef {Record<string, any> & { now: () => number }} PerformanceLike
+ */
+
+/**
  * @typedef {object} Timers
  * @property {SetTimeout} setTimeout - native `setTimeout`
  * @property {ClearTimeout} clearTimeout - native `clearTimeout`
@@ -318,7 +322,7 @@ if (typeof require === "function" && typeof module === "object") {
  * @property {ClearImmediate} [clearImmediate] - native `clearImmediate`, if available
  * @property {Hrtime} [hrtime] - native `process.hrtime`, if available
  * @property {NextTick} [nextTick] - native `process.nextTick`, if available
- * @property {Performance} [performance] - native `performance`, if available
+ * @property {PerformanceLike} [performance] - native `performance`, if available
  * @property {RequestAnimationFrame} [requestAnimationFrame] - native `requestAnimationFrame`, if available
  * @property {QueueMicrotask} [queueMicrotask] - whether `queueMicrotask` exists
  * @property {CancelAnimationFrame} [cancelAnimationFrame] - native `cancelAnimationFrame`, if available

--- a/test/typescript-no-dom-consumer/consume-types.ts
+++ b/test/typescript-no-dom-consumer/consume-types.ts
@@ -1,0 +1,16 @@
+import FakeTimers from "@sinonjs/fake-timers";
+
+declare global {
+    class AbortSignal {}
+}
+
+const timers = FakeTimers.timers;
+
+if (timers.performance) {
+    const now: number = timers.performance.now();
+    now.toFixed();
+}
+
+// @ts-expect-error DOM globals should not be available in this consumer.
+const windowRef: Window = {};
+windowRef;

--- a/test/typescript-no-dom-consumer/tsconfig.json
+++ b/test/typescript-no-dom-consumer/tsconfig.json
@@ -1,0 +1,17 @@
+{
+  "compilerOptions": {
+    "lib": ["ES2022"],
+    "module": "Node16",
+    "target": "ESNext",
+    "moduleResolution": "node16",
+    "strict": true,
+    "skipLibCheck": false,
+    "esModuleInterop": true,
+    "noEmit": true,
+    "types": [],
+    "paths": {
+      "@sinonjs/fake-timers": ["../../types/fake-timers-src.d.ts"]
+    }
+  },
+  "include": ["consume-types.ts"]
+}

--- a/types/fake-timers-src.d.ts
+++ b/types/fake-timers-src.d.ts
@@ -95,6 +95,9 @@ export type SetTickModeConfig = {
 export type IntlWithClock = Record<string, any> & {
     clock: Clock;
 };
+export type PerformanceLike = Record<string, any> & {
+    now: () => number;
+};
 export type Timers = {
     setTimeout: SetTimeout;
     clearTimeout: ClearTimeout;
@@ -106,7 +109,7 @@ export type Timers = {
     clearImmediate?: ClearImmediate;
     hrtime?: Hrtime;
     nextTick?: NextTick;
-    performance?: Performance;
+    performance?: PerformanceLike;
     requestAnimationFrame?: RequestAnimationFrame;
     queueMicrotask?: QueueMicrotask;
     cancelAnimationFrame?: CancelAnimationFrame;

